### DISCUSSION TITLE
How to use exclusions correctly

### DISCUSSION THREAD

**Discussion by isaac3203** [brianpetro/obsidian-smart-connections#77] [General]
I want to prevent the journal files in my ob library from being retrieved by smart-connection, since they have little knowledge. I've tried path exclusions and folder exclusions, but neither seem to work. Remarks, I am a plug-in user who uses Chinese. For example, my diary is stored in the folder "Diary Book" in the ob library. Can you give me an example demonstration, how to exclude it?

**Comment by brianpetro** [comment]
Simply adding "Diary Book" to the exclusions, separated by a comma from any other entries, should do the trick. However, it doesn't delete existing Embeddings, which could be leading to the confusion.

Deleting excluded files/folders could be a good feature request to add to the issues list. 

In the meantime, unfortunately, you will have to do a force refresh, which deletes all Embeddings, after adding the exclusion if you want them to be deleted.

**Reply by pinuke** [reply]
Just out of curiosity, do exclusions support globbing? Like as in the npm package globby? It may be worth adding that feature, if it isn't already implemented.

**Reply by brianpetro** [reply]
No globbing at the moment. I'm holding off depending on any NPM packages as long as possible. So just a simple matcher for now.

**Reply by pinuke** [reply]
You and your fear of IoC lol. You're killing me smalls. Ha ha.

Though, I do look forward to it. I appreciate the amount of time you invest into this plugin.

**Reply by brianpetro** [reply]
What can I say? I like simple.

**Reply by zeroliu** [reply]
Slightly related question, if I reorganize my folder structure, will it update previously generated embeddings?

**Reply by brianpetro** [reply]
@zeroliu yes, that will trigger re-embedding because the location of the note within folders is included in the information that is embedded.

**Comment by quelsemme** [comment]
For some reason, file exclusion isn't working for me.

I want to exclude all files with this word in the title: outline, draft, template, notes
I want to exclude these folders:
- 0001 Notes/00017 Planning Notes
- 0000 Inbox



**I have tried these options and nothing is working:**

file_exclusions: outline, draft, template, notes
header_exclusions: outline, draft, template, notes

**Option #1:** folder_exclusions: "0001 Notes/00017 Planning Notes/", "0000 Inbox/"
**Option #2:** folder_exclusions: 0001 Notes/00017 Planning Notes, 0000 Inbox
**Option #3:** folder_exclusions: 00017 Planning Notes, 0000 Inbox


After each attempt, I refreshed the embeddings to make sure it wasn't working off an old version of the vault. Can you advise which is the correct syntax to make this work for me?

**Reply by quelsemme** [reply]
I want it to exclude all the sub-folders of those mentioned folders too, btw.

**Comment by PhilMProctor** [comment]
I'm also having issues with excluding folders.
I add folders I wish to exclude e.g. Journal/Daily Notes,Excalidraw
I run a force_refresh and now folders which are **not** excluded are showing as excluded.
I can only get access to those folders again if I remove all entries in the excluded folders
I'm running version 1.6.53.

Am I using the wrong syntax i.e. should each folder be in quotes?

**Reply by nomadphase** [reply]
I am having some issues with exclusions, with my daily notes folder appearing within the embeddings despite being excluded.

I wonder if a different UI, where one selected from a drop-down list (or similar) of folders for exclusion might be more intuitive.

Also, is it feasible to ask whether a file should remain excluded when moving it from one folder to another?

**Reply by brianpetro** [reply]
@nomadphase 

RE: maintaining exclusion after moving:

What do you think about exclusions based on a tag or front matter?

That would probably be a much easier implementation and would persist regardless of where the file was located.

RE: the improved exclusion settings:

There is a GitHub issue for this, which you should give a 👍 to help me keep track of the priority: https://github.com/brianpetro/obsidian-smart-connections/issues/389 

🌴

**Comment by ryran** [comment]
> RE: maintaining exclusion after moving:
> 
> What do you think about exclusions based on a tag or front matter?
> 
> That would probably be a much easier implementation and would persist regardless of where the file was located.

Yes! That gets my vote. I haven't even done my initial generation of embeddings; I've only installed the plugin and was trying to figure out how/whether I could exclude things via front-matter properties.
